Assignment 2.Rmd

```diff
@@ -87,14 +91,100 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.widt
 
 In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
 
+```{r}
+teachers=data.frame(EDGE %>%
+                      group_by(from) %>%
+                      summarise(count = n()))
+teachers
+
+students<-count(group_by(EDGE,to))
+students
+
+#students<-summarise(group_by(EDGE,to),count=n())
+
+names(teachers)=c("characters","count")
+teachers
+
+names(students)=c("characters","count")
+students
+
+VERTEX2<-dplyr::bind_rows(teachers,students)
+VERTEX2
+
+library(igraph)
+
+g2=graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX2)
+g2
+plot(g2,layout=layout.fruchterman.reingold, vertex.color = VERTEX$gender, vertex.size = (VERTEX2$count)*5)
+
+```
 
 
 ## Part III
 
 Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
 
+```{r}
+HUDK4050=read.csv("/Users/mac/Desktop/assignment2/hudk4050-classes.csv")
+HUDK4050=data.frame(HUDK4050)
+library(igraph)
+library(tidyr)
+library(dplyr)
+NewHUDK4050=data.frame(HUDK4050) %>% 
+  unite(Name,First.Name,Last.Name,remove=TRUE,sep=" ")
+
+class<-NewHUDK4050 %>% gather(key=class,value=course_code,c(Class.1,Class.2,Class.3,Class.4,Class.5,Class.6)) %>% filter(!is.na(course_code))
+class<-class[,-2]
+
+class$course_code <- gsub(pattern = " ",replacement = "", x = class$course_code)
+class[34,2]<-"HUDK4050"
+class[87,2]<-"HUDM4125"
+class[140,2]<-"HUDM5026"
+class[151,2]<-"G5072"
+class[164,2]<-"G5072"
+class[193,2]<-"HUDM5126"
+class[234,2]<-"G5015"
+
+countclass<-ifelse(class$course_code=="",class$count<-0,class$count<-1)
+classes<-class[!class$course_code == "",]
+class<-unique(class)
+
+classes<-spread(classes,course_code,count,fill=0)
+matrix1<-as.matrix(classes)
+row.names(matrix1)<-classes$Name
+matrix1<-matrix1[,-1]
+matrix1<-apply(matrix1,2,as.numeric)
+matrix2<-matrix1%*%t(matrix1)
+row.names(matrix2)<-classes$Name
+matrix1<-matrix1[,-1]
+matrix1<-apply(matrix1,2,as.numeric)
+matrix2<-matrix1%*%t(matrix1)
+row.names(matrix2)<-classes$Name
+colnames(matrix2)<-classes$Name
+diag(matrix2)<-NA #Aviod problem
+
+
+#Get the graph data
+graph_data<-graph.adjacency(matrix2,mode="undirected")
+
+#Plot the data
+V(graph_data)$color<-ifelse(degree(graph_data)==max(degree(graph_data)), "red", "blue")
+plot(graph_data,layout=layout.fruchterman.reingold,vertex.label.cex=0.5,vertex.size=12)
```

Comment:
You have done a good job generating your graph, though since you did not remove HUDK4050 from the data all nodes are connected to each other and it looks like one big ball.